pkg/subctl/cmd/gather.go

```diff
@@ -0,0 +1,170 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/submariner-operator/pkg/internal/cli"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/components"
+)
+
+var (
+	gatherType   string
+	gatherModule string
+)
+
+const (
+	Logs = "logs"
+	Resources = "resources"
+)
+var gatherModuleFlags = map[string]bool{
+	components.Connectivity:     false,
+	components.ServiceDiscovery: false,
+	components.Broker:           false,
+	components.Operator:         false,
```

Comment:
Wouldn’t it make sense to distinguish Globalnet too, since it has its own agent?

Reply:
> Wouldn’t it make sense to distinguish Globalnet too, since it has its own agent?
It can fall under connectivity, we can easily detect if globalnet is enabled and get those logs too.